File: Python/src/maths/pow.py
"""TO FIX: NEGATIVE Y """

def pow(x: int, y: int) -> int:
    """
    Calculates the result of raising x to the power y (x^y)
    
    Args:
    - x (int): The base number.
    - y (int): The exponent.
    
    Returns:
    - int: The result of x raised to the power y.
    """
    result: int = 1
    
    for i in range(0, y):
        result *= x
        
    return result

def pow_recursive(x: int, y: int) -> int:
    if (y == 0): 
        return 1
    
    return x * pow(x, y - 1)

File: C++/main/maths/pow.cpp
#include <iostream>
/**
 * Calculates the power of a number (x^y) using iterative method.
 *
 * @param x The base number.
 * @param y The exponent.
 * @return The result of raising x to the power y (x^y)
 */
long pow(int x, int y)
{
    long result = 1;
    for (int i = 0; i < y; i++)
    {
        result *= x;
    }
    return result;
}
/**
 * Calculates the power of a number (x^y) using recursive method.
 *
 * @param x The base number.
 * @param y The exponent.
 * @return The result of raising x to the power y (x^y)
 */
long pow_recursive(int x, int y)
{
    if (y == 0)
        return 1;
    return x * pow_recursive(x, y - 1);
}

int main()
{
    // Some examples:

    int base = 2;
    int exponent = 3;
    std::cout << " Iterative Power: " << pow(base, exponent) << std::endl;
    std::cout << "Recursive Power: " << pow_recursive(base, exponent) << std::endl;

    base = 10;
    exponent = 5;
    std::cout << "Iterative Power: " << pow(base, exponent) << std::endl;
    std::cout << "Recursive Power: " << pow_recursive(base, exponent) << std::endl;

    base = 27;
    exponent = 5;
    std::cout << "Iterative Power: " << pow(base, exponent) << std::endl;
    std::cout << "Recursive Power: " << pow_recursive(base, exponent) << std::endl;
    return 0;
}